www/forms.py

```diff
@@ -82,16 +82,56 @@ class FileImportForm(forms.Form):
     )
     file = forms.FileField()
 
+class CustomInvoiceServiceChoiceField(forms.ModelChoiceField):
+    """
+    Get better selection text for services in custom invoices.
+    """
+    def label_from_instance(self, obj):
+        return _("%(servicename)s, %(cost)s") % {
+            "servicename": obj.service.name,
+            "cost": obj.service.cost_string(),
+        }
+
 
 class CustomInvoiceForm(forms.Form):
-    service = forms.ModelChoiceField(
+    service = CustomInvoiceServiceChoiceField(
         label=_("Service"),
         queryset=ServiceSubscription.objects.none(),
         empty_label=_("Choose service.."),
     )
     count = forms.IntegerField(
         label=_("How many units of service you want"), min_value=1, max_value=666
     )
+    price = forms.DecimalField(
+        label=_("Price per unit"),
+        help_text=_("See price from the service selected above")
+    )
+
+    def clean(self):
+        """
+        Super clean
+        And check that the price matches with the selected service
+        """
+        cleaned_data = super().clean()
+        subscription = cleaned_data["service"]
+        service = subscription.service
+        price = cleaned_data["price"]
+        if(service.cost_max and price > service.cost_max):
+            raise forms.ValidationError(
+                _('Price cannot be above service max cost: %(max)s'),
+                params={'max': service.cost_max},
```

Comment:
Needs adding some sanity checking against some minimum real money value, obviously can't be zero or negative, so "over zero" would propably be exactly correct.